pymc/model_graph.py

```diff
@@ -102,11 +102,31 @@ def get_parents(self, var: TensorVariable) -> Set[VarName]:
         parents = self._get_ancestors(var, func)
         return self._filter_parents(var, parents)
 
-    def make_compute_graph(self) -> Dict[str, Set[VarName]]:
+    def vars_to_plot(self, selected_vars=None):
+        if selected_vars is None:
+            return self.var_names
+        else:
+            all_selected_ancestors = set()
+
+            for selected_var_name in selected_vars:
+                var_ancestors = self._get_ancestors(
+                    self.model[selected_var_name], self.model[selected_var_name]
+                )
+                all_selected_ancestors = all_selected_ancestors.union(var_ancestors)
+
+            vars_on_graph = all_selected_ancestors.union(
+                {self.model[selected_var_name] for selected_var_name in selected_vars}
+            )
+            vars_on_graph = {var.name for var in vars_on_graph}
+
+            # ordering of self.var_names is important
+            return [var_name for var_name in self.var_names if var_name in vars_on_graph]
+
+    def make_compute_graph(self, selected_vars=None) -> Dict[str, Set[VarName]]:
         """Get map of var_name -> set(input var names) for the model"""
         input_map = defaultdict(set)  # type: Dict[str, Set[VarName]]
 
-        for var_name in self.var_names:
+        for var_name in self.vars_to_plot(selected_vars):
```

Comment:
Commenting here because the diff above has all these codecov warnings...

In addition to the minimal example, I tested it with my big model and the result was missing a lot of variables.
The ancestor determination in `vars_to_plot()` appears to miss a lot of nodes.

The old code started from `self.var_names` which appears to be the complete list. Maybe the way how it's determined in `__init__` is relevant?